### ISSUE TITLE
conftest.c(38): error: identifier "Syntax" is undefined

### ISSUE THREAD

**Issue by simon-Leary42** [Unidata/netcdf-c#2207]
### Environment Information
* What platform are you using? (please provide specific distribution/version in summary)
  * [x] Linux
  * [ ] Windows
  * [ ] OSX
  * [ ] Other
  * [ ] NA
* 32 and/or 64 bit?
  * [ ] 32-bit
  * [x] 64-bit
* What build system are you using?
  * [x] autotools (`configure`)
  * [ ] cmake
* Can you provide a sample netCDF file or `C` code to recreate the issue?
  * [ ] Yes (please attach to this issue, thank you!)
  * [x] No
  * [ ] Not at this time

### Summary of Issue
config.log -> https://pastebin.com/3TRxafqf
error message -> https://pastebin.com/F3DzzYAi
./configure tries to compile "Syntax Error" as c code.
This occurs in something called confdefs.h, but I can find so such file in the source code.
```
| # include <limits.h>
| #else
| # include <assert.h>
| #endif
| 		     Syntax error
conftest.c(38): error: identifier "Syntax" is undefined
  		     Syntax error
  		     ^
conftest.c(38): error: expected a ";"
```

I think that upon failure ./configure deletes confdefs.h because it doesn't exist.

I found that ./configure has 427 occurrences of "confdefs.h" and 127 occurrences of this line:
`cat confdefs.h - <<_ACEOF >conftest.$ac_ext`

I'm at a loss. What can I do to track this down?

### Steps to reproduce the behavior
```
#!/bin/bash
NCDIR="/modules/apps/netcdf-c/4.8.1_intel"

module load icc

cd netcdf-c-4.8.1
./configure --prefix=${NCDIR} --disable-netcdf-4 --disable-dap
```

**Comment by DennisHeimbigner** [comment]
The confdefs.h is created multiple times to test for various platform features
(like testing for some header file, etc).
So, the real problem here is why configure is finding a syntax error.
Let's start by having you send us the config.log file.

**Comment by simon-Leary42** [comment]
> 

there is a link to a pastebin of config.log in my post

**Comment by DennisHeimbigner** [comment]
Sorry, I have no idea how to fix this. It appears that the code in configure is
wrong, which means that your autotools is not operating correctly.
You might try re-installing automake and autoconf, but otherwise,
I have idea how to find the fault.

**Comment by simon-Leary42** [comment]
I used cmake instead and it worked just fine.

**Comment by simon-Leary42** [comment]
> Sorry, I have no idea how to fix this. It appears that the code in configure is wrong, which means that your autotools is not operating correctly. You might try re-installing automake and autoconf, but otherwise, I have idea how to find the fault.

I didn't generate configure with autoconf, this configure file came with the release

**Comment by DennisHeimbigner** [comment]
True. But it works for everyone else. There is something in your environment/tool-set
that is not working correctly and ./configure is invoking that tool and it is failing.
Lets try this:
Execute ./configure using the command
````sh -x ./configure <whatever options you are using>````
and send me the output. Warning it will be very large.

**Comment by simon-Leary42** [comment]
> True. But it works for everyone else. There is something in your environment/tool-set that is not working correctly and ./configure is invoking that tool and it is failing. Lets try this: Execute ./configure using the command `sh -x ./configure <whatever options you are using>` and send me the output. Warning it will be very large.

https://pastebin.com/TpV41kJ7

**Comment by simon-Leary42** [comment]
> True. But it works for everyone else. There is something in your environment/tool-set that is not working correctly and ./configure is invoking that tool and it is failing. Lets try this: Execute ./configure using the command `sh -x ./configure <whatever options you are using>` and send me the output. Warning it will be very large.

https://controlc.com/81d6fad8
I don't think pastebin likes 11,000 lines...

**Comment by DennisHeimbigner** [comment]
Can you compress it with gzip or bzip2 or zip?

**Comment by simon-Leary42** [comment]
[debug.zip](https://github.com/Unidata/netcdf-c/files/7974424/debug.zip)

**Comment by DennisHeimbigner** [comment]
As near as I can tell, the problem is with the icc c-preprocessor.
But it also appears that the output of sh -x ./configure you sent is truncated,
so I cannot see what is causing the failure.
[Ward do we have an installation of icc?]

**Comment by simon-Leary42** [comment]
I fixed the issue by changing the $CPP environment variable from "icc -E" to "icpc -E"

**Comment by DennisHeimbigner** [comment]
Glad to hear it; sorry I could not be any help.
